ocrd_anybaseocr/cli/ocrd_anybaseocr_dewarp.py

```diff
@@ -1,184 +1,185 @@
+            page_grp = self.output_file_grp
+            file_id = input_file.ID.replace(self.input_file_grp, page_grp)
             if file_id == input_file.ID:
-                file_id = concat_padded(page_grp, n)                
+                file_id = concat_padded(page_grp, n)
             self.workspace.add_file(
                 ID=file_id,
                 file_grp=page_grp,
                 pageId=input_file.pageId,
                 mimetype=MIMETYPE_PAGE,
-                local_filename=os.path.join(page_grp,
-                                        file_id + '.xml'),
+                local_filename=os.path.join(page_grp, file_id + '.xml'),
                 content=to_xml(pcgts).encode('utf-8'),
                 force=self.parameter['force']
             )
-        os.rmdir(self.input_file_grp+"/test_A/") #FIXME: better way of deleting a temp_dir?
-        
 
     def _process_segment(self, model, dataset, page, page_xywh, page_id, input_file, orig_img_size, n):
-        for i, data in enumerate(dataset):
-            w,h = orig_img_size
-            generated = model.inference(data['label'], data['inst'], data['image'])
-            dewarped = array(generated.data[0].permute(1,2,0).detach().cpu())
-            bin_array = array(255*(dewarped>ocrolib.midrange(dewarped)),'B')
+        for _, data in enumerate(dataset):
+            w, h = orig_img_size
+            generated = model.inference(
+                data['label'], data['inst'], data['image'])
+            dewarped = array(generated.data[0].permute(1, 2, 0).detach().cpu())
+            bin_array = array(255*(dewarped > ocrolib.midrange(dewarped)), 'B')
             dewarped = ocrolib.array2pil(bin_array)
-            dewarped = dewarped.resize((w,h))                        
-            
-            page_xywh['features'] += ',dewarped'  
-            
-            file_id = input_file.ID.replace(self.input_file_grp, self.image_grp)
+            dewarped = dewarped.resize((w, h))
+
+            page_xywh['features'] += ',dewarped'
+
+            file_id = input_file.ID.replace(self.input_file_grp, self.output_file_grp)
             if file_id == input_file.ID:
-                file_id = concat_padded(self.image_grp, n)
-        
-            file_path = self.workspace.save_image_file(dewarped,
-                                   file_id,
-                                   page_id=page_id,
-                                   file_grp=self.image_grp,
-                                   force=self.parameter['force']
-                )     
-            page.add_AlternativeImage(AlternativeImageType(filename=file_path, comments=page_xywh['features']))
-        
+                file_id = concat_padded(self.output_file_grp, n)
 
+            file_path = self.workspace.save_image_file(dewarped,
+                                                       file_id,
+                                                       page_id=page_id,
```

Comment:
Note: `page_id` defined at the start of `process` can only be used for logging, as it has `input_file.ID` as a fallback.

```suggestion
                                                       page_id=input_file.pageId,
```